modules/config/i18n.nix

```diff
@@ -0,0 +1,44 @@
+# The glibc package in nixpkgs is patched to make it possbile to specify
+# an alternative path for the locale archive through a special environment
+# variable. This would allow different versions of glibc to coexist on the
+# same system because each version of glibc could look up different paths
+# for its locale archive should the archive format ever change in
+# incompatible ways.
+#
+# See also:
+# localedef(1)
+# https://nixos.org/manual/nixpkgs/stable/#locales
+# https://github.com/NixOS/nixpkgs/issues/38991
+#
+# XXX: The name of the said environment variable gets updated with each
+# breaking release of the glibcLocales package. Periodically check the link
+# below for changes:
+# https://github.com/NixOS/nixpkgs/blob/nixpkgs-unstable/pkgs/development/libraries/glibc/nix-locale-archive.patch
+
+{ lib, pkgs, ... }:
+
+with lib;
+
+let
+  inherit (pkgs.glibcLocales) version;
+
+  archivePath = "${pkgs.glibcLocales}/lib/locale/locale-archive";
+
+  # lookup the version of glibcLocales and set the appropriate environment vars
+  localeVars = if (versionAtLeast version "2.27") then {
+    LOCALE_ARCHIVE_2_27 = archivePath;
+  } else if (versionAtLeast version "2.11") then {
+    LOCALE_ARCHIVE_2_11 = archivePath;
+  } else
+    { };
+in {
+  config = {
+    # for shell sessions
+    home.sessionVariables = localeVars;
+
+    # for desktop apps
+    systemd.user.sessionVariables = localeVars;
```

Comment:
Given this is gated behind `isLinux`, would it make sense to add the locale vars to `pam.sessionVariables`?

Reply:
I didn't add it given that reading user environments with pam_env is [deprecated][1],  and from what I can gather, is already disabled on major distros. I guess we can add it if many users still rely on it, but it's likely going to stop working sooner or later.

[1]: https://github.com/linux-pam/linux-pam/commit/ecd526743a27157c5210b0ce9867c43a2fa27784

Reply:
In that case, there is little use in adding it as a PAM session variable